### ISSUE TITLE
2D categorical plots do not plot all categories 

### ISSUE THREAD

**Issue by operte** [histogrammar/histogrammar-python#46]
In the basic notebook tutorial there is an example on a 2D categorical plot.
The raw data contains 3 x 5 categories, but only 2 x 4 bins are visible when plotting with matplotlib:

```python
histy = hg.Categorize('eyeColor', )
histx = hg.Categorize('favoriteFruit', value=histy)
histx.fill.numpy(df)
histx.plot.matplotlib();
```

<img width="457" alt="image" src="https://user-images.githubusercontent.com/4223965/162387167-32e72612-3b29-4286-be95-7d9db7fa9470.png">

The values seem to be correctly stored in the histogram, just not correctly plotted:

```python
x_labels, y_labels, grid = get_2dgrid(histx)
x_labels, y_labels, grid 

(['apple', 'banana', 'strawberry'],
 ['blue', 'brown', 'green', 'grey', 'red'],
 array([[24., 27., 26.],
        [25., 24., 22.],
        [34., 25., 23.],
        [21., 29., 26.],
        [39., 28., 27.]]))
```

**Comment by operte** [comment]
After digging a bit, it seems the problem lies with `pcolormesh`, used in https://github.com/histogrammar/histogrammar-python/blob/dd3be584e9b55a3a1a9e9c3bad2c7ed5e075d2b1/histogrammar/plot/matplotlib.py#L763

<img width="555" alt="image" src="https://user-images.githubusercontent.com/4223965/162394992-d440babd-854e-4bce-a117-fbbeceb66f07.png">

<img width="559" alt="image" src="https://user-images.githubusercontent.com/4223965/162395039-299ca8fd-945d-4233-8743-01185d2f4101.png">

It seems `pcolormesh` doesn't like categorical coordinates. Forcing `shading='nearest'` doesn't change anything.

**Comment by operte** [comment]
It seems the issue is during the plotting with `pcolormesh` in line https://github.com/histogrammar/histogrammar-python/blob/dd3be584e9b55a3a1a9e9c3bad2c7ed5e075d2b1/histogrammar/plot/matplotlib.py#L763

`pcolormesh` expects to receive n+1 edge coordinates, but for categorical variables it gets only n.

If lines 759 to 761 are changed to
```python
# plot 2d grid
xtick_pos = np.arange(len(x_labels)) + 0.5
ytick_pos = np.arange(len(y_labels)) + 0.5
```

I believe that the categorical features are plotted correctly, but one would have to check if nothing wrong happens to the numerical ones.

**Comment by mbaak** [comment]
Thanks Joao!

I got it fixed now. Indeed pcolormesh needs edges, not centers. 
I'll push the fix and release a new patch release.

**Comment by mbaak** [comment]
Fyi I released histogrammar v1.0.26 with the fix.
